deployment/stacks/api.ts

```diff
@@ -107,142 +130,99 @@ export function createApiComponents(props: CreateApiComponentsProps) {
     SAR_FORM_BUCKET: sarFormBucket.bucketName,
     ABCD_FORM_BUCKET: abcdFormBucket.bucketName,
     ...Object.fromEntries(
-      tables.map((table) => [`${table.id}Table`, table.name])
+      tables.map((table) => [`${table.node.id}Table`, table.table.tableName])
     ),
   };
 
-  const additionalPolicies = [
-    new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: [
-        "dynamodb:DescribeTable",
-        "dynamodb:Query",
-        "dynamodb:Scan",
-        "dynamodb:GetItem",
-        "dynamodb:PutItem",
-        "dynamodb:UpdateItem",
-        "dynamodb:DeleteItem",
-      ],
-      resources: [
-        ...tables.map((table) => table.arn),
-        ...tables.map((table) => `${table.arn}/index/*`),
-      ],
-    }),
-    new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: ["s3:GetObject", "s3:ListBucket", "s3:PutObject"],
-      resources: [
-        `${wpFormBucket.bucketArn}/formTemplates/*`,
-        wpFormBucket.bucketArn,
-        `${wpFormBucket.bucketArn}/formTemplates/*`,
-        `${wpFormBucket.bucketArn}/fieldData/*`,
-        sarFormBucket.bucketArn,
-        `${sarFormBucket.bucketArn}/formTemplates/*`,
-        `${sarFormBucket.bucketArn}/fieldData/*`,
-        abcdFormBucket.bucketArn,
-        `${abcdFormBucket.bucketArn}/formTemplates/*`,
-        `${abcdFormBucket.bucketArn}/fieldData/*`,
-      ],
-    }),
-
-    new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: [
-        "dynamodb:DescribeStream",
-        "dynamodb:GetRecords",
-        "dynamodb:GetShardIterator",
-        "dynamodb:ListShards",
-        "dynamodb:ListStreams",
-      ],
-      resources: tables.map((table) => table.streamArn).filter(isDefined),
-    }),
-    new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: ["s3:GetObject", "s3:PutObject", "s3:ListBucket"],
-      resources: [
-        wpFormBucket.bucketArn,
-        sarFormBucket.bucketArn,
-        abcdFormBucket.bucketArn,
-        `${wpFormBucket.bucketArn}/fieldData/*`,
-        `${sarFormBucket.bucketArn}/fieldData/*`,
-        `${abcdFormBucket.bucketArn}/fieldData/*`,
-      ],
-    }),
-  ];
-
   const commonProps = {
     stackName: `${service}-${stage}`,
     api,
     environment,
-    additionalPolicies,
+    tables,
     isDev,
   };
 
-  new Lambda(scope, "createBanner", {
+  const createBanner = new Lambda(scope, "createBanner", {
```

Comment:
Any chance we could set this up like seds so that we don't have to manage the permissions of each lambda granularly?

Reply:
Yes, we can do that.  The reason I did it this way here is so we don't give a function that doesn't need it, read/write access to buckets and tables.  Principle of least privilege.